### ISSUE TITLE
Fails to find custom service account

### ISSUE THREAD

**Issue by hawksight** [terraform-google-modules/terraform-google-kubernetes-engine#1160]
### TL;DR

When choosing not to auto create a service account, but instead pass reference to my own account, the plan works but at application time the cluster fails to create claiming the service account cannot be found.

### Expected behavior

I'd expect that the creation would work as normal, because the account exists and I even made the `gke` module depend on the service account creation.

### Observed behavior

Instead I get an error returned as such:

```
╷
│ Error: googleapi: Error 400: Service account "projects/<PROJECT>/serviceAccounts/gke-demo@<PROJECT>.iam.gserviceaccount.com" does not exist., badRequest
│
│   with module.gke.google_container_cluster.primary,
│   on .terraform/modules/gke/cluster.tf line 22, in resource "google_container_cluster" "primary":
│   22: resource "google_container_cluster" "primary" {
│
```

The service account exists and can be seen in the UI.

### Terraform Configuration

```hcl
variable "region" {
  type        = string
  description = "The region to host the cluster in"
  default     = "europe-west2"
}

variable "zones" {
  type        = list(string)
  description = "The zone to host the cluster in (required if is a zonal cluster)"
  default     = ["europe-west2-b"]
}

variable "ip_range_pods" {
  description = "The secondary ip range to use for pods"
  default     = "192.168.0.0/18"
}

variable ip_range_pods_name {
  type        = string
  default     = "pod-ips"
  description = "Name of the secondary ip range for pods"
}

variable "ip_range_services" {
  description = "The secondary ip range to use for services"
  default     = "192.168.64.0/18"
}

variable ip_range_services_name {
  type        = string
  default     = "service-ips"
  description = "Name of the secondary ip range for services"
}

variable release_name {
  type        = string
  default     = "demo"
  description = "The installation name, kind of like a helm release name. All resource naming based on this"
}

variable "project_id" {
  type        = string
  description = "The project ID to host the cluster in"
}

locals {
  release = lower(trim(var.release_name, " "))
}

data "google_client_config" "default" {}

resource "google_service_account" "gke" {
  project      = var.project_id
  account_id   = "gke-${local.release}" 
  display_name = "GKE Service Account - ${var.release_name}"
}

# -- Networking -- #
module "gcp-network" {
  source  = "terraform-google-modules/network/google"
  version = ">= 4.0.1, < 5.0.0"

  project_id   = var.project_id
  network_name = "vpc-${local.release}"

  subnets = [
    {
      subnet_name           = "sub-${local.release}-1"
      subnet_ip             = "10.0.0.0/17"
      subnet_region         = var.region
      subnet_private_access = "true"
    },
  ]

  secondary_ranges = {
    ("sub-${local.release}-1") = [
      {
        range_name    = var.ip_range_pods_name
        ip_cidr_range = var.ip_range_pods
      },
      {
        range_name    = var.ip_range_services_name
        ip_cidr_range = var.ip_range_services
      },
    ]
  }
}

data "google_compute_subnetwork" "subnetwork" {
  name       = "sub-${local.release}-1"
  project    = var.project_id
  region     = var.region
  depends_on = [module.gcp-network]
}

# -- Cluster Resources -- #
module "gke" {
  source     = "terraform-google-modules/kubernetes-engine/google"
  version    = ">=19.0.0"

  project_id = var.project_id
  name       = "gke-${local.release}"
  regional   = false
  region     = var.region
  zones      = slice(var.zones, 0, 1)

  network                 = module.gcp-network.network_name
  subnetwork              = module.gcp-network.subnets_names[0]
  ip_range_pods           = var.ip_range_pods_name
  ip_range_services       = var.ip_range_services_name
  create_service_account  = false
  release_channel         = "RAPID"

  master_authorized_networks = [
    {
      cidr_block   = data.google_compute_subnetwork.subnetwork.ip_cidr_range
      display_name = "VPC"
    },
  ]

  # Node leve settiongs
  remove_default_node_pool = true

  node_pools = [
    {
      name                      = "np-1-${local.release}"
      machine_type              = "e2-medium"
      node_locations            = element(var.zones, 0)
      min_count                 = 1
      max_count                 = 100
      local_ssd_count           = 0
      disk_size_gb              = 20
      disk_type                 = "pd-standard"
      image_type                = "COS_CONTAINERD"
      auto_repair               = true
      auto_upgrade              = true
      service_account           = google_service_account.gke.id
      preemptible               = false
      initial_node_count        = 1
    },
  ]

  node_pools_oauth_scopes = {
    all = [ "https://www.googleapis.com/auth/cloud-platform"]
  }

  node_pools_labels = {
    all = {}
  }

  node_pools_metadata = {
    all = {}
  }

  node_pools_taints = {
    all = []
  }

  node_pools_tags = {
    all = []
  }

  depends_on = [module.gcp-network, google_service_account.gke]
}
```

### Terraform Version

```sh
Terraform v1.1.6
on darwin_amd64
+ provider registry.terraform.io/hashicorp/external v2.2.0
+ provider registry.terraform.io/hashicorp/google v4.11.0
+ provider registry.terraform.io/hashicorp/google-beta v4.11.0
+ provider registry.terraform.io/hashicorp/kubernetes v2.8.0
+ provider registry.terraform.io/hashicorp/null v3.1.0
+ provider registry.terraform.io/hashicorp/random v3.1.0
```

### Additional information

I'm trying to work out if there is an issue specifically with my project, or if this is a reproducible error.
It worked using `create_service_account= true` but I was waning to pre-create my service account for visibility in this instance. Perhaps I'm missing an API that needs enabled to lookup the service account?

**Comment by morgante** [comment]
Please use the email.

**Comment by hawksight** [comment]
It was the `.email` that I needed, thanks.
